### PR TITLE
combine galley startmode and liveness/readiness probe

### PR DESCRIPTION
**Pull request by irisdingbj**

### CODE REVIEW

**Comment by irisdingbj** [comment]
@ayj same code change, and I removed the circleci from my own repo, now the testing running on istio branch and ci/circle:test passed.

File: install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml [hunk 1]
```diff
@@ -26,22 +26,24 @@ spec:
...
           - /usr/local/bin/galley
-          - validator
-          - --deployment-namespace={{ .Release.Namespace }}
           - --caCertFile=/etc/istio/certs/root-cert.pem
           - --tlsCertFile=/etc/istio/certs/cert-chain.pem
           - --tlsKeyFile=/etc/istio/certs/key.pem
-          - --healthCheckInterval=1s
-          - --healthCheckFile=/health
-          - --webhook-config-file
+          - --livenessProbeInterval=2s
```
Comment (ayj) [inline install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml]: Can you restore the liveness and readiness probe intervals to their original values?

```
--readinessProbeInterval=1s
--livenessProbeInterval=1s

--interval=10s
--initialDelaySeconds: 5
periodSeconds=5
```

File: galley/pkg/server/server.go [hunk 2]
```diff
@@ -211,16 +194,40 @@ func (s *Server) Close() error {
...
+		serverReadinessProbe.RegisterProbe(readinessProbeController, "serverReadiness")
+		defer serverReadinessProbe.SetAvailable(errors.New("stopped"))
+	}
+
+	err = s.Wait()
+	if err != nil {
+		fatalf("Galley Server unexpectedly terminated: %v", err)
+	}
+	_ = s.Close()
+
```
Comment (ayj) [inline galley/pkg/server/server.go]: nit: extra newline

File: galley/pkg/server/controller.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	if livenessProbeController != nil {
+		livenessProbeController.Start()
+		defer livenessProbeController.Close()
+	}
+	if readinessProbeController != nil {
+		readinessProbeController.Start()
+		defer readinessProbeController.Close()
+	}
+	<-stop
+
```
Comment (ayj) [inline galley/pkg/server/controller.go]: extra newline

File: install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml (hunk 1 above)
Comment (cmluciano) [reply install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml]: @irisdingbj please address these with a new PR

File: galley/pkg/server/args.go [hunk 4]
```diff
@@ -50,27 +49,23 @@ type Args struct {
...
-	LivenessProbeOptions *probe.Options
-
-	// The path to the file for readiness probe, similar to LivenessProbePath.
-	ReadinessProbeOptions *probe.Options
-
 	// The introspection options to use
 	IntrospectionOptions *ctrlz.Options
+
+	// Enable running galley server mode
+	EnableServer bool
```
[the remark this answers is not in the document]
Comment (irisdingbj) [reply galley/pkg/server/args.go]: done

File: galley/pkg/crd/validation/webhook.go [hunk 5]
```diff
@@ -107,6 +94,9 @@ type WebhookParameters struct {
 	DeploymentName string
 
 	Clientset clientset.Interface
+
+	// Enable running galley validation mode
```
[the remark this answers is not in the document]
Comment (irisdingbj) [reply galley/pkg/crd/validation/webhook.go]: done

File: galley/cmd/galley/cmd/root.go [hunk 6]
```diff
@@ -48,21 +64,88 @@ func GetRootCmd(args []string, printf, fatalf shared.FormatFn) *cobra.Command {
...
+	rootCmd.PersistentFlags().StringVar(&livenessProbeOptions.Path, "livenessProbePath", server.DefaultLivenessProbeFilePath,
+		"Path to the file for the Galley liveness probe.")
+	rootCmd.PersistentFlags().DurationVar(&livenessProbeOptions.UpdateInterval, "livenessProbeInterval", server.DefaultProbeCheckInterval,
+		"Interval of updating file for the Galley liveness probe.")
+	rootCmd.PersistentFlags().StringVar(&readinessProbeOptions.Path, "readinessProbePath", server.DefaultReadinessProbeFilePath,
+		"Path to the file for the Galley readiness probe.")
+	rootCmd.PersistentFlags().DurationVar(&readinessProbeOptions.UpdateInterval, "readinessProbeInterval", server.DefaultProbeCheckInterval,
+		"Interval of updating file for the Galley readiness probe.")
+	rootCmd.PersistentFlags().UintVar(&monitoringPort, "monitoringPort", 9093,
+		"Port to use for the exposing self-monitoring information")
```
[the remark this answers is not in the document]
Comment (irisdingbj) [reply galley/cmd/galley/cmd/root.go]: done

File: install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml (hunk 1 above)
Comment (irisdingbj) [reply install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml]: done

File: galley/pkg/server/server.go (hunk 2 above)
Comment (irisdingbj) [reply galley/pkg/server/server.go]: done

File: galley/pkg/server/controller.go (hunk 3 above)
Comment (irisdingbj) [reply galley/pkg/server/controller.go]: done

**Pull request merged**